redoctober.go

```diff
@@ -203,6 +204,21 @@ func (this *indexHandler) handle(w http.ResponseWriter, r *http.Request) {
 	http.ServeContent(w, r, "index.html", time.Now(), body)
 }
 
+// initPrometheus starts a goroutine with a Prometheus listener that
+// listens on localhost:metricsPort. If the Prometheus handler can't
+// be started, a log.Fatal call is made.
+func initPrometheus() {
+	srv := &http.Server{
+		Addr:    net.JoinHostPort("localhost", metricsPort),
+		Handler: prometheus.Handler(),
+	}
+
+	log.Printf("metrics.init start: addr=%s", srv.Addr)
+	go func() {
+		log.Fatal(srv.ListenAndServe())
```

Comment:
Ah. So the code I copied that from, which does work, puts it in a goroutine, like so:
```
go serveMetrics()
func serveMetrics() {
  log.Fatal(http.ListenAndServe(*metricsAddr, prometheus.Handler()))
}
```
Not sure why that's necessary, but the fact that ListenAndServe doesn't return unless there's an error is probably it.